jobs/datalake-data-export/src/services/read-model-queries.service.ts

```diff
@@ -13,21 +13,51 @@ export class ReadModelQueriesService {
 
   public async getEServices(): Promise<Array<ExportedEService>> {
     return this.readModel.eservices
-      .find()
+      .find({
+        $or: [
+          {
+            'data.descriptors.1': { $exists: true },
+          },
+          {
+            'data.descriptors': {
+              $size: 1,
+              $elemMatch: {
+                state: { $in: ['Published', 'Archived', 'Suspended', 'Deprecated'] satisfies DescriptorState[] },
+              },
+            },
+          },
+        ],
+      })
       .map(({ data }) => ExportedEService.parse(data))
       .toArray()
   }
 
   public async getAgreements(): Promise<Array<ExportedAgreement>> {
     return this.readModel.agreements
-      .find()
+      .find({
+        'data.state': { $ne: 'Draft' satisfies AgreementState },
+      })
       .map(({ data }) => ExportedAgreement.parse(data))
       .toArray()
   }
 
   public async getPurposes(): Promise<Array<ExportedPurpose>> {
     return this.readModel.purposes
-      .find()
+      .find({
+        $or: [
+          {
+            'data.versions.1': { $exists: true },
+          },
+          {
+            'data.versions': {
+              $size: 1,
+              $elemMatch: {
+                state: { $in: ['Active', 'Archived', 'Suspended'] satisfies PurposeState[] },
+              },
+            },
+          },
+        ],
+      })
```

Comment:
A purpose can be deleted if has no versions, or has one with "Draft"/"WaitingForApproval" state.
We need to filter them out from the result.